frontend/src/components/Header/UserInfo.js

```diff
@@ -13,29 +16,37 @@ const BOOTSTRAP_SPACING_CLASS =
  * log in), the user's display name, and a Sign Out button. This component is to
  * be used with the Header component.
  */
-class UserInfo extends React.Component {
-  /** @inheritdoc */
-  render() {
-    return (
-      <div className='d-flex flex-row'>
-        <img
-            className={BOOTSTRAP_SPACING_CLASS}
-            src={authUtils.getUserPhotoUrl()}
-            alt='Your Profile'
-        />
-        <p className={BOOTSTRAP_SPACING_CLASS}>
-          {authUtils.getUserDisplayName()}
-        </p>
-        <Button
-            className={BOOTSTRAP_SPACING_CLASS}
-            variant='secondary'
-            onClick={authUtils.signOut}
-        >
-          Sign Out
-        </Button>
-      </div>
-    );
+const UserInfo = () => {
+  const history = useHistory();
+
+  /**
+   * Redirects the user to the LANDING page first before actually signing them
+   * out.
+   */
+  function signOutAndRedirectToLanding() {
+    history.push(LANDING);
+    authUtils.signOut();
   }
```

Comment:
Why does this redirect to the landing page? Also, why is the order redirect -> log out?

Reply:
Right now, the way authentication works is that whenever the website notices that the user is not logged in, it'll immediately redirect them to the signin page. This is so if a user tries to access a private page like View Trips but they aren't logged in, they can quickly sign in once they are redirected to the sign in page.

I'm assuming most users would expect to be redirected to the landing/home page when they press the "Sign Out" button instead of being redirected to the signin page. So I first tell the browser to bring the user to the home page before logging them out. If I did it the other way around, the user would first be brought to the sign in page, then brought to the home page immediately after, which might look strange if there are any delays.